ts/components/Pinpad/index.tsx

```diff
@@ -162,7 +162,7 @@ class Pinpad extends React.PureComponent<Props, State> {
       const isValid = inputValue === this.props.compareWithCode;
 
       if (!isValid && this.props.clearOnInvalid) {
-        this.clear();
+        setTimeout(() => this.clear(), 100);
```

Comment:
It works!
Btw you could try to use [debounce](https://lodash.com/docs/4.17.15#debounce) function
The advantage to use that function is that the clear function will be invoked until after wait milliseconds have elapsed since the last time the debounced function was invoked.

Using _setTimeout_ this could happen:
- users insert an invalid value
- **_clear_** will be invoked after 100 milliseconds
- if user makes a changes before 100 milliseconds elapsed (he is Flash :running:), **_clear_** will be invoked anyway
- user loses his changes

with _debounce_ we could prevent this behavior

Reply:
updated within https://github.com/teamdigitale/io-app/pull/1301/commits/a4c50571e41cb1fcccac32a12612a75e59437d58